### PR TITLE
Allow object spaced curlies

### PR DESCRIPTION
**Pull request by jhhayashi**

### CODE REVIEW

**Comment by billyjanitsch** [comment]
We use [babel-eslint](https://github.com/babel/babel-eslint) (instead of the default parser used by the Airbnb config) so that ESLint can parse stage 1 syntax. This causes some native rules to break, so the parser has an [accompanying plugin](https://github.com/babel/eslint-plugin-babel#rules) which provides replacements for those rules. See [this part](https://github.com/kensho/eslint-config-kensho/blob/master/index.js#L4) of our config.

tl;dr to make this change, you need to keep the native object-curly-spacing disable and update the params of the babel replacement to align with the Airbnb config.

**Comment by billyjanitsch** [comment]
Once we switch to using prettier for style formatting, this change will be [a lot easier](https://github.com/kensho/eslint-config-kensho/pull/18/files#diff-168726dbe96b3ce427e7fedce31bb0bcR51).
